### PR TITLE
go: env/actions/branch: Move and delete branch also update working sets.

### PR DESCRIPTION
**Pull request by reltuk**

### CODE REVIEW

File: integration-tests/bats/branch.bats [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    dolt gc
+    new_sz=`du -s | awk '{print $1}'`
+    dolt checkout master
+    dolt branch -d -f to_delete
+    num_branches=`dolt branch | wc -l`
+    [[ "$num_branches" -eq 1 ]] || fail "expected num_branches to be 1"
+    dolt gc
+    post_delete_sz=`du -s | awk '{print $1}'`
+    echo "$sz $new_sz $post_delete_sz"
+    [[ "$post_delete_sz" -eq "$sz" ]] || false
```
Comment (reltuk) [inline integration-tests/bats/branch.bats:34]: Maybe this is too restrictive for Windows...

**Comment by VinaiRachakonda** [comment]
This was a fun read!

File: go/libraries/doltcore/doltdb/doltdb.go [hunk 2]
```diff
@@ -890,6 +890,31 @@ func (ddb *DoltDB) NewBranchAtCommit(ctx context.Context, branchRef ref.DoltRef,
 	return ddb.UpdateWorkingSet(ctx, wsRef, ws, currWsHash, nil)
 }
 
+func (ddb *DoltDB) CopyWorkingSet(ctx context.Context, fromWSRef ref.WorkingSetRef, toWSRef ref.WorkingSetRef, force bool) error {
+	ws, err := ddb.ResolveWorkingSet(ctx, fromWSRef)
+	if err == ErrWorkingSetNotFound {
+		return nil
```
Comment (zachmu) [inline go/libraries/doltcore/doltdb/doltdb.go:896]: Method docs that note this quirk in behavior plz

**Review by zachmu — APPROVED** [review]
LGTM!

**Pull request merged**